### DISCUSSION TITLE
Fastlane iOS :  sh: xcpretty: command not found

### DISCUSSION THREAD

**Discussion by jyotsanas2020** [fastlane/fastlane#18149] [Q&A]
```
Getting error while building app through Fastlane :  sh: xcpretty: command not found
+---------------------------------------------------------------+----------------------------------------------------------------+
|                                                    Summary for gym 2.173.0                                                     |
+---------------------------------------------------------------+----------------------------------------------------------------+
| export_method                                                 | app-store                                                      |
| export_options.provisioningProfiles.com.kpit.viessmanndemoapp | *********                                                 |
| project                                                       | ./DemoApp.xcodeproj                                       |
| scheme                                                        | DemoApp                                                   |
| destination                                                   | generic/platform=iOS                                           |
| output_name                                                   | *************                                       |
| build_path                                                    | /Users/abc/Library/Developer/Xcode/Archives/2021-02-07 |
| clean                                                         | false                                                          |
| output_directory                                              | .                                                              |
| silent                                                        | false                                                          |
| skip_package_ipa                                              | false                                                          |
| skip_package_pkg                                              | false                                                          |
| result_bundle                                                 | false                                                          |
| buildlog_path                                                 | ~/Library/Logs/gym                                             |
| skip_profile_detection                                        | false                                                          |
| skip_package_dependencies_resolution                          | false                                                          |
| disable_package_automatic_updates                             | false                                                          |
| use_system_scm                                                | false                                                          |
| xcode_path                                                    | /Applications/Xcode.app                                        |
+---------------------------------------------------------------+----------------------------------------------------------------+

+------------------+-------------+
|          Lane Context          |
+------------------+-------------+
| DEFAULT_PLATFORM | ios         |
| PLATFORM_NAME    | ios         |
| LANE_NAME        | ios release |
+------------------+-------------+
[17:10:44]: Error building the application - see the log above

+------+------------------+-------------+
|           fastlane summary            |
+------+------------------+-------------+
| Step | Action           | Time (in s) |
+------+------------------+-------------+
| 1    | default_platform | 0           |
| 💥   | build_app        | 291         |
+------+------------------+-------------+
```

ERROR [2021-02-08 08:21:17.66]: fastlane finished with errors
Traceback (most recent call last):
	34: from /usr/local/bin/fastlane:23:in `<main>'
	33: from /usr/local/bin/fastlane:23:in `load'
	32: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/bin/fastlane:23:in `<top (required)>'
	31: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
	30: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	29: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/commands_generator.rb:352:in `run'
	28: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	27: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	26: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	25: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	24: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	23: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	22: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	21: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
	20: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	19: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	18: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	17: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	16: from Fastfile:25:in `block (2 levels) in parsing_binding'
	15: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	14: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	13: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	12: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	11: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	10: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	 9: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	 8: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/actions/build_app.rb:68:in `run'
	 7: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/gym/lib/gym/manager.rb:19:in `work'
	 6: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/gym/lib/gym/runner.rb:18:in `run'
	 5: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/gym/lib/gym/runner.rb:110:in `build_app'
	 4: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane_core/lib/fastlane_core/command_executor.rb:90:in `execute'
	 3: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/gym/lib/gym/runner.rb:114:in `block in build_app'
	 2: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/gym/lib/gym/error_handler.rb:73:in `handle_build_error'
	 1: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
/usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane_core/lib/fastlane_core/ui/interface.rb:163:in `build_failure!': Error building the application - see the log above (FastlaneCore::Interface::FastlaneBuildFailure)
	34: from /usr/local/bin/fastlane:23:in `<main>'
	33: from /usr/local/bin/fastlane:23:in `load'
	32: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/bin/fastlane:23:in `<top (required)>'
	31: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
	30: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	29: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/commands_generator.rb:352:in `run'
	28: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	27: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	26: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	25: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	24: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	23: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	22: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	21: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
	20: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	19: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	18: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	17: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	16: from Fastfile:25:in `block (2 levels) in parsing_binding'
	15: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	14: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	13: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	12: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	11: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	10: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	 9: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	 8: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane/lib/fastlane/actions/build_app.rb:68:in `run'
	 7: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/gym/lib/gym/manager.rb:19:in `work'
	 6: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/gym/lib/gym/runner.rb:18:in `run'
	 5: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/gym/lib/gym/runner.rb:110:in `build_app'
	 4: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane_core/lib/fastlane_core/command_executor.rb:90:in `execute'
	 3: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/gym/lib/gym/runner.rb:114:in `block in build_app'
	 2: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/gym/lib/gym/error_handler.rb:73:in `handle_build_error'
	 1: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
/usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.173.0/fastlane_core/lib/fastlane_core/ui/interface.rb:163:in `build_failure!': \e[31m[!] Error building the application - see the log above\e[0m (FastlaneCore::Interface::FastlaneBuildFailure

**Comment by dinsen** [comment]
Hi @jyotsanas2020 
Could you provide us with the output of `fastlane env`?

**Reply by jyotsanas2020** [reply]
<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 11.1                                        |
| Ruby                        | 2.7.2                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.24.3 (Apple Git-128)          |
| Installation Source         | /usr/local/bin/fastlane                     |
| Host                        | macOS 11.1 (20C69)                          |
| Ruby Lib Dir                | /usr/local/Cellar/ruby@2.7/2.7.2/lib        |
| OpenSSL Version             | OpenSSL 1.1.1i  8 Dec 2020                  |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 12.4                                        |


### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |


### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:ios)

platform :ios do
  desc "Push a new release build to the App Store"
  lane :release do
unlock_keychain(
    path: "/Users/maximilianr/Library/Keychains/login.keychain-db",
    password: "welcome@123"
  )
    	  build_app(
    export_options: {
      provisioningProfiles: { 
        "com.kpit.viessmanndemoapp"=>"MaximilianRenk"
      }
    }
  )
   	 #upload_to_app_store
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier("com.kpit.viessmanndemoapp") # The bundle identifier of your app
apple_id("jyotsanas@kpit.com") # Your Apple email address

itc_team_id("559558") # App Store Connect Team ID
team_id("QYBDGVLQ8B") # Developer Portal Team ID

# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.173.0 | ✅ Up-To-Date  |


### Loaded fastlane plugins:

**No plugins Loaded**


<details><summary><b>Loaded gems</b></summary>

| Gem                           | Version      |
| ----------------------------- | ------------ |
| did_you_mean                  | 1.4.0        |
| slack-notifier                | 2.3.2        |
| atomos                        | 0.1.3        |
| CFPropertyList                | 3.0.3        |
| claide                        | 1.0.3        |
| colored2                      | 3.1.2        |
| nanaimo                       | 0.3.0        |
| xcodeproj                     | 1.19.0       |
| rouge                         | 2.0.7        |
| xcpretty                      | 0.3.0        |
| terminal-notifier             | 2.0.0        |
| unicode-display_width         | 1.7.0        |
| terminal-table                | 1.8.0        |
| plist                         | 3.6.0        |
| public_suffix                 | 4.0.6        |
| addressable                   | 2.7.0        |
| multipart-post                | 2.0.0        |
| word_wrap                     | 1.0.0        |
| tty-screen                    | 0.8.1        |
| tty-cursor                    | 0.7.1        |
| tty-spinner                   | 0.9.3        |
| artifactory                   | 3.0.15       |
| babosa                        | 1.0.4        |
| colored                       | 1.2          |
| highline                      | 1.7.10       |
| commander-fastlane            | 4.4.6        |
| excon                         | 0.78.1       |
| faraday-net_http              | 1.0.1        |
| faraday                       | 1.3.0        |
| unf_ext                       | 0.0.7.7      |
| unf                           | 0.1.4        |
| domain_name                   | 0.5.20190701 |
| http-cookie                   | 1.0.3        |
| faraday-cookie_jar            | 0.0.7        |
| faraday_middleware            | 1.0.0        |
| fastimage                     | 2.2.1        |
| gh_inspector                  | 1.1.3        |
| mini_magick                   | 4.11.0       |
| rubyzip                       | 2.3.0        |
| security                      | 0.1.3        |
| xcpretty-travis-formatter     | 1.0.1        |
| dotenv                        | 2.7.6        |
| bundler                       | 2.1.4        |
| simctl                        | 1.6.8        |
| jwt                           | 2.2.2        |
| uber                          | 0.1.0        |
| declarative                   | 0.0.20       |
| declarative-option            | 0.1.0        |
| representable                 | 3.0.4        |
| retriable                     | 3.1.2        |
| mini_mime                     | 1.0.2        |
| multi_json                    | 1.15.0       |
| signet                        | 0.14.0       |
| memoist                       | 0.16.2       |
| os                            | 1.1.1        |
| googleauth                    | 0.14.0       |
| httpclient                    | 2.8.3        |
| google-api-client             | 0.38.0       |
| emoji_regex                   | 3.2.1        |
| aws-eventstream               | 1.1.0        |
| aws-sigv4                     | 1.2.2        |
| aws-sdk-kms                   | 1.41.0       |
| aws-sdk-s3                    | 1.87.0       |
| uri                           | 0.10.0       |
| ruby2_keywords                | 0.0.4        |
| json                          | 2.5.1        |
| naturally                     | 2.2.1        |
| rake                          | 13.0.3       |
| digest-crc                    | 0.6.3        |
| webrick                       | 1.7.0        |
| rexml                         | 3.2.4        |
| google-apis-core              | 0.2.1        |
| google-apis-storage_v1        | 0.1.0        |
| google-apis-iamcredentials_v1 | 0.1.0        |
| google-cloud-errors           | 1.0.1        |
| google-cloud-env              | 1.4.0        |
| google-cloud-core             | 1.5.0        |
| google-cloud-storage          | 1.30.0       |
| aws-partitions                | 1.418.0      |
| jmespath                      | 1.4.0        |
| aws-sdk-core                  | 3.111.2      |
| forwardable                   | 1.3.1        |
| logger                        | 1.4.2        |
| cgi                           | 0.1.0        |
| date                          | 3.0.0        |
| timeout                       | 0.1.0        |
| stringio                      | 0.1.0        |
| zlib                          | 1.1.0        |
| ipaddr                        | 1.2.2        |
| openssl                       | 2.1.2        |
| ostruct                       | 0.2.0        |
| strscan                       | 1.0.3        |
| delegate                      | 0.1.0        |
| fileutils                     | 1.4.1        |
| io-console                    | 0.5.6        |
| open3                         | 0.1.0        |
| singleton                     | 0.1.0        |
| yaml                          | 0.1.0        |
| psych                         | 3.1.0        |
| mutex_m                       | 0.1.0        |
</details>


*generated on:* **2021-02-08**
</details>
